crates/bevy_ptr/src/lib.rs

```diff
@@ -1103,13 +1103,17 @@ impl<'a, T> From<&'a [T]> for ThinSlicePtr<'a, T> {
 }
 
 /// Creates a dangling pointer with specified alignment.
-/// See [`NonNull::dangling`].
+///
+/// This is an untyped version of [`NonNull::dangling()`]. The returned pointer has no
+/// [provenance](https://doc.rust-lang.org/stable/std/ptr/index.html#provenance).
+///
+/// # Panics
+///
+/// If `align` is not a power of two (2, 4, 8, 16, etc.).
```

Comment:
I would not put this in the documentation if it panics only when debug assertions are enabled.

Reply:
I disagree as I think undocumented panics are not user-friendly, even if they only panic with certain build configurations.

Either way, since I ended up deleting this line anyways, I'm going to mark this as resolved.

Reply:
> I disagree as I think undocumented panics are not user-friendly, even if they only panic with certain build configurations.

In that case I would at least document that this happens only when debug assertions and that it is not guaranteed in general. There is a difference IMO between "may panic if" and "surely panics if"